call-home/src/bin/callhome/main.rs

```diff
@@ -208,12 +220,34 @@ async fn generate_report(
 
     let replicas = http_client.replicas_api().get_replicas().await;
     match replicas {
-        Ok(replicas) => report.replicas = Replicas::new(replicas.into_body().len(), volumes),
-        Err(err) => {
+        Ok(ref replicas) => {
+            report.replicas = Replicas::new(replicas.clone().into_body().len(), volumes)
+        }
+        Err(ref err) => {
             error!("{:?}", err);
         }
     };
 
+    if let Ok(pools) = pools {
+        report.spdk_managed_disks = SpdkManagedDisks::new(pools.clone().into_body(), disks.clone());
+        if let Ok(replicas) = replicas {
+            report.storage_nodes =
+                StorageNodes::new(replicas.clone().into_body(), pools.into_body())
+        }
+    }
+
+    // find valid disks to calculate storage media metrics
+    let valid_disks = disks
+        .into_iter()
+        .filter(|device| {
+            device.size > 0
+                && device.devtype != "partition"
```

Comment:
what happens if the mayastor pool is created on a partition?

Reply:
I am considering only the physical disk devices to calculate the storage media metrics. And the physical disks and partitions used by pools to calculate spdk managed disk metrics.

Reply:
@tiagolobocastro  - This Feature was request by PM for deeper understanding of users storage infra
What we need to understand is more info about the disk media allocated to the cluster itself. How Mayastor utilizes that disk media is important (full disk or GPT partitioned);  but this doesn't help us understand how a users storage capacity is evolving within the cluster. It actually confuses our understanding of it. Looking at the entire disk give us an exact understanding.